src/pages/Home/__tests__/index.test.tsx

```diff
@@ -0,0 +1,255 @@
+        is_required: false,
+        is_known: true,
+      },
+    },
+    {
+      key: 0,
+      value: {
+        name: "data-loss-protect",
+        is_required: true,
+        is_known: true,
+      },
+    },
+    {
+      key: 7,
+      value: {
+        name: "gossip-queries",
+        is_required: false,
+        is_known: true,
+      },
+    },
+    {
+      key: 23,
+      value: {
+        name: "anchors-zero-fee-htlc-tx",
+        is_required: false,
+        is_known: true,
+      },
+    },
+    {
+      key: 27,
+      value: {
+        name: "shutdown-any-segwit",
+        is_required: false,
+        is_known: true,
+      },
+    },
+    {
+      key: 30,
+      value: {
+        name: "amp",
+        is_required: true,
+        is_known: true,
+      },
+    },
+    {
+      key: 31,
+      value: {
+        name: "amp",
+        is_required: false,
+        is_known: true,
+      },
+    },
+    {
+      key: 2023,
+      value: {
+        name: "script-enforced-lease",
+        is_required: false,
+        is_known: true,
+      },
+    },
+    {
+      key: 5,
+      value: {
+        name: "upfront-shutdown-script",
+        is_required: false,
+        is_known: true,
+      },
+    },
+  ],
+};
+
+function createEvent(eventName: string, eventContent: unknown) {
+  return `event: ${eventName}\ndata: ${JSON.stringify(eventContent)}\n\n`;
+}
+
+const encoder = new TextEncoder();
+
+// Create MSW server
+server.use(
+  http.get("/api/lightning/list-all-tx", () => {
+    return HttpResponse.json([]);
+  }),
+  http.get(SSE_URL, () => {
+    const stream = new ReadableStream({
+      start(controller) {
+        // Simulate SSE messages
+        controller.enqueue(
+          encoder.encode(createEvent("system_startup_info", systemStartupInfo)),
+        );
+
+        controller.enqueue(
+          encoder.encode(createEvent("wallet_balance", walletBalance)),
+        );
+
+        controller.enqueue(
+          encoder.encode(createEvent("system_info", systemInfo)),
+        );
+
+        controller.enqueue(encoder.encode(createEvent("ln_info", lnInfo)));
+
+        controller.enqueue(encoder.encode(createEvent("ln_info", lnInfo)));
+
+        controller.close();
+      },
+    });
+
+    return new HttpResponse(stream, {
+      headers: {
+        "Content-Type": "text/event-stream",
+        Connection: "keep-alive",
+        "Cache-Control": "no-cache",
+      },
+    });
+  }),
+);
+
+describe("Home Component", () => {
+  it("renders and handles SSE updates correctly", async () => {
+    render(
+      <BrowserRouter>
+        <I18nextProvider i18n={i18n}>
+          <AppContextProvider>
+            <SSEContextProvider>
+              <Layout>
+                <Home />
+              </Layout>
+            </SSEContextProvider>
+          </AppContextProvider>
+        </I18nextProvider>
+      </BrowserRouter>,
+    );
+
+    await waitFor(() => {
+      const totalBalanceValue = screen
```

Comment:
Could we just do the same, we do in the unit-tests, i.e.
`expect(await screen.findByText("123,456 SAT")).toBeInTheDocument();`

Adding dataTestIds is not good practice I believe. That's why we avoided these in the unit-tests as well.

Reply:
> Adding dataTestIds is not good practice I believe. That's why we avoided these in the unit-tests as well.

Yes, was just frustrated because I didn't get the result I wanted at first :sweat_smile: 

> Could we just do the same, we do in the unit-tests, i.e.
expect(await screen.findByText("123,456 SAT")).toBeInTheDocument();

Agreed!
Since the messaging mock now works, I would do tests for the component (e.g. the balance test for the WalletCard) & in the Home I could check for the higher-level stuff, e.g. when does the loading spinner appear etc.